src/app/Plans/components/helpers.ts

```diff
@@ -1,33 +1,133 @@
   const condition = plan.status?.conditions.find(
     (condition) =>
-      condition.type === PlanStatusType.Ready ||
-      condition.type === PlanStatusType.Executing ||
-      condition.type === PlanStatusType.Succeeded ||
-      condition.type === PlanStatusType.Failed
+      condition.type === 'Ready' ||
+      condition.type === 'Executing' ||
+      condition.type === 'Succeeded' ||
+      condition.type === 'Failed'
   );
   return condition ? PlanStatusDisplayType[condition.type] : '';
 };
 
-// TODO maybe generalize this for cold migrations too
-type WarmPlanState =
-  | 'NotStarted'
-  | 'Starting'
-  | 'Copying'
-  | 'AbortedCopying'
-  | 'StartingCutover'
-  | 'Cutover'
-  | 'Finished';
+export const getMigStatusState = (plan: IPlan) => {
+  const conditions = plan.status?.conditions || [];
+  let title = '';
+  let variant: ProgressVariant | undefined;
+
+  switch (true) {
+    case hasCondition(conditions, 'Executing') && plan.spec.warm: {
+      title = 'Running cutover';
+      break;
+    }
+    case hasCondition(conditions, 'Executing'): {
+      title = 'Running';
+      break;
+    }
+    case hasCondition(conditions, 'Succeeded'): {
+      title = 'Succeeded';
+      variant = ProgressVariant.success;
+      break;
+    }
+    case hasCondition(conditions, 'Canceled'): {
+      title = 'Canceled';
+      break;
+    }
+    case hasCondition(conditions, 'Failed'): {
+      title = 'Failed';
+      variant = ProgressVariant.danger;
+      break;
+    }
+    case !!plan.status?.migration?.started: {
+      if (canPlanBeStarted(plan)) {
+        title = 'Finished - Incomplete';
+        variant = ProgressVariant.warning;
+      } else {
+        console.warn('Migration plan unexpected status:', plan);
+      }
+      break;
+    }
+    default: {
+      title = '';
+    }
+  }
+
+  return {
+    title,
+    variant,
+  };
+};
 
-export const getWarmPlanState = (
+export const getButtonState = (
+  state: PlanState | null,
+  plan: IPlan,
+  migration: IMigration | null
+): PlanActionButtonType | null => {
+  let type: PlanActionButtonType | null = null;
+
+  switch (true) {
+    case state === 'NotStarted': {
+      type = 'Start';
+      break;
+    }
+    case plan.status?.migration?.started && canPlanBeStarted(plan): {
+      type = 'Restart';
+      break;
+    }
+    case state === 'Starting': {
+      type = null; // spinner
+      break;
+    }
+    case state === 'Starting' && !migration?.spec.cutover: {
+      type = 'Cutover'; // spinner
+      break;
+    }
+    case state === 'Copying': {
+      type = 'Cutover';
+      break;
+    }
+    case state === 'AbortedCopying': {
+      type = 'Restart';
+      break;
+    }
+    case state === 'StartingCutover': {
+      type = null; // spinner
+      break;
+    }
+    // "Cutover" name is warm-specific, but this state applies to both warm and cold
+    // it could just be called 'Migrating' or 'PipelineRunning'
+    case state === 'Cutover': {
```

Comment:
I think we should go ahead and rename `Cutover` so it's not warm-specific, that could get confusing if we forget to do it later.